src/libraries/System.Runtime/tests/System.Runtime.Tests/System/TimeZoneInfoTests.cs

```diff
@@ -2316,6 +2316,31 @@ public static void NJulianRuleTest(string posixRule, int dayNumber, int monthNum
             }
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public static void ArbitraryTZ_UsedAsLocal()
+        {
+            const string tzId = "America/Monterrey";
+            const string tzPath = "/usr/share/zoneinfo/" + tzId;
+
+            if (!File.Exists(tzPath))
+            {
+                throw new SkipTestException($"The file {tzPath} does not exist.");
+            }
+            
+            RemoteExecutor.Invoke(() =>
+            {
+                string tmp = Path.GetTempPath() + Path.GetRandomFileName();
+                File.WriteAllBytes(tmp, File.ReadAllBytes(tzPath));
+
+                Environment.SetEnvironmentVariable("TZ", tmp);
+                TimeZoneInfo tzi = TimeZoneInfo.GetSystemTimeZones().FirstOrDefault(t => t.Id == tzId);
```

Comment:
You either do that before calling the remote executor and pass the environment variable setting (look at NJulianRuleTest for how we do that). Or, you clear the cache before calling `TimeZoneInfo.GetSystemTimeZones`.